### ISSUE TITLE
[Feature] Support more detailed errors within Python generator in JSON parsing

### ISSUE THREAD

**Issue by armandobelardo** [fern-api/fern#3069]
### Problem description

Today, in the event the JSON response cannot be parsed to the corresponding Pydantic object, we throw generic Pydantic errors, ex:

```
pydantic.error_wrappers.ValidationError: 1 validation error for ParsingModel[EntitySummaryResponse]
__root__ -> attributes -> address -> data -> 1 -> properties -> type
```

This is nice IF you have the data and can see the problematic data you're receiving, however in this instance, Pydantic does not return the actual problematic data and so you have to have stored the JSON elsewhere to analyze it and see what's causing issues.

This could go hand in hand with broader improvements to debugging (think: being able to log the full sent request, loggging the response, etc.)

### Why would it be useful?

You could imagine support teams at companies asking users to flip on detailed debugging or other feature on their client, should there be an issue to be able to send detailed logs.
In general, being able to provide more insight into why clients are not responding as they should (e.g. getting a look into the requests and responses being sent) could be pretty valuable

**Comment by dannysheridan** [comment]
@eyw520 this feels like it could be up your alley!

**Comment by eyw520** [comment]
Absolutely. Error logging is something @dsinghvi and I have had plenty of headaches dealing with haha. Will be a great thing to improve.
